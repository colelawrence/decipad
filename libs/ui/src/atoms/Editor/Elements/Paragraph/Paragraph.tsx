import { css } from '@emotion/react';
import { PlatePluginComponent } from '@udecode/plate';
import { p16Regular } from '../../../../primitives';

const styles = css(p16Regular, {
  padding: '8px 0',
});

export const ParagraphElement: PlatePluginComponent = ({
  attributes,
  children,
  nodeProps,
}) => {
  // TODO type root better and make optional
  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
  const { root } = nodeProps!.styles;
  return (
    <p css={[styles, root.css]} {...attributes} {...nodeProps}>
      {children}
    </p>
  );
};
