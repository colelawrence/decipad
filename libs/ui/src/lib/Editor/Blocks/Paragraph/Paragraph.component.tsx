import { Text } from '@chakra-ui/react';
import React from 'react';
import { RenderElementProps } from 'slate-react';

export const Paragraph = ({
  attributes,
  children,
}: RenderElementProps): JSX.Element => {
  return (
    <Text as="p" {...attributes}>
      {children}
    </Text>
  );
};
