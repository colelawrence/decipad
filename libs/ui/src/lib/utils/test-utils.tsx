import { ChakraProvider } from '@chakra-ui/react';
import { render, RenderOptions } from '@testing-library/react';
import React from 'react';
import { theme } from '../theme';

const AllTheProviders: React.FC = ({ children }) => {
  return (
    <ChakraProvider resetCSS theme={theme}>
      {children}
    </ChakraProvider>
  );
};

const customRender = (
  ui: JSX.Element,
  options?: Omit<RenderOptions, 'queries'>
) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';
// override render method
export { customRender as render };
