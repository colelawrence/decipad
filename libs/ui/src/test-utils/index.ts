export * from './menu';
export * from './language';
