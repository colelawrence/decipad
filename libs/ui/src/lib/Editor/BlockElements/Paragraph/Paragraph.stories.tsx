import React from 'react';
import { Paragraph } from './Paragraph.component';

export default {
  title: 'Editor/Block Elements/Paragraph',
};

export const P = () => (
  <Paragraph
    element={null}
    attributes={{
      'data-slate-node': 'element',
      ref: null,
    }}
  >
    The quick brown fox jumps over the lazy dog
  </Paragraph>
);

P.storyName = 'Paragraph';
