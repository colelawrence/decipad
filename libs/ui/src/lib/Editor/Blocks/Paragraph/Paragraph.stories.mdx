import { Meta, Story, Canvas, Props } from '@storybook/addon-docs/blocks';
import { Paragraph } from './Paragraph.component';

<Meta
  title="Legacy/Editor/Blocks/Paragraph"
  parameters={{
    docs: {
      source: {
        code: `<Paragraph
  element={{
    children: [{ text: 'Hello World' }],
  }}
  attributes={{ 'data-slate-node': 'element', ref: null }}
>
  Hello World
</Paragraph>`,
      },
    },
  }}
  argTypes={{
    children: {
      defaultValue: 'Hello World',
      description: 'The text content of the component',
      type: 'string',
    },
  }}
  component={Paragraph}
/>

# Paragraph

The Paragraph component is the most basic slate component that we have in the editor, it is just a normal `p` tag basically and it is the block that is used for normal text.

export const Template = (args) => <Paragraph {...args} />;

<Canvas>
  <Story name="Paragraph">{Template.bind({})}</Story>
</Canvas>

<Props story="Paragraph" />
