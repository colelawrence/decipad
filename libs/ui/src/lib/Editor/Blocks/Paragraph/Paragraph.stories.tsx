import React from 'react';
import { Paragraph } from './Paragraph.component';

export default {
  title: 'Editor/Block Elements/Paragraph',
};

export const P = () => (
  <Paragraph
    element={{ children: [{ text: 'hello world ' }] }}
    attributes={{
      'data-slate-node': 'element',
      ref: null,
    }}
  >
    The quick brown fox jumps over the lazy dog
  </Paragraph>
);

P.storyName = 'Paragraph';
