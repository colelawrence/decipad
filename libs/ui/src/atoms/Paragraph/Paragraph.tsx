import { css } from '@emotion/react';
import { ReactNode } from 'react';
import { cssVar, p16Regular, setCssVar } from '../../primitives';
import { SlateElementProps } from '../../utils';

const placeholderStyles = css({
  position: 'relative',

  '::before': {
    ...p16Regular,
    ...setCssVar('currentTextColor', cssVar('weakTextColor')),
    cursor: 'text',
    pointerEvents: 'none',

    position: 'absolute',

    content: 'attr(aria-placeholder)',
  },
});

const styles = css(p16Regular, placeholderStyles, {
  padding: '6px 0',
});

interface ParagraphProps extends SlateElementProps {
  readonly children: ReactNode;
  /**
   * Note: Since this is not a plain-text element like an `<input>`,
   * it is up to the consumer to ensure the placeholder is removed
   * when there is content in the children that it could overlap with.
   */
  readonly placeholder?: string;
}

export const Paragraph = ({
  children,
  placeholder,
  slateAttrs,
}: ParagraphProps): ReturnType<React.FC> => {
  return (
    <p aria-placeholder={placeholder} css={styles} {...slateAttrs}>
      {children}
    </p>
  );
};
