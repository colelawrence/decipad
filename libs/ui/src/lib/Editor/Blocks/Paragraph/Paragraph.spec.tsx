import { render } from 'test-utils';
import { Paragraph } from './Paragraph.component';

describe('Paragraph Block', () => {
  it('renders', () => {
    const { container, getByText } = render(
      <Paragraph
        element={{
          children: [{ text: 'Hello World' }],
        }}
        attributes={{ 'data-slate-node': 'element', ref: null }}
      >
        Hello World
      </Paragraph>
    );

    expect(container).toBeDefined();
    getByText('Hello World');
  });

  it('matches snapshot', () => {
    const { container } = render(
      <Paragraph
        element={{
          children: [{ text: 'Hello World' }],
        }}
        attributes={{ 'data-slate-node': 'element', ref: null }}
      >
        Hello World
      </Paragraph>
    );

    expect(container).toMatchSnapshot();
  });
});
