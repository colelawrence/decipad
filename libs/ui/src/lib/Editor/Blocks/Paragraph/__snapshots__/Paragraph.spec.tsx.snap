// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph Block matches snapshot 1`] = `
<div>
  <p
    class="chakra-text emotion-0"
    data-slate-node="element"
  >
    Hello World
  </p>
</div>
`;
